### ISSUE TITLE
Failed to parse csv with cyrillic values

### ISSUE THREAD

**Issue by sairus7** [JuliaData/CSV.jl#1034]
I have a strange bug that never appeared before on the same workflows with the same CSV.jl version.

```julia
str = """record	number
rec1	2166
rec2	2166
rec3	2155
GROSS	20760
Average	20760"""
io = IOBuffer(str)
df = CSV.File(io, delim = '\t') # this works fine

str = """record	number
rec1	2166
rec2	2166
rec3	2155
GROSS	20760
Среднее	20760"""
io = IOBuffer(str)
df = CSV.File(io, delim = '\t') #error
```
 The last line gives me error:
```
ERROR: MethodError: Cannot `convert` an object of type
  Parsers.Result{String15} to an object of type 
  Parsers.Result{Any}
Closest candidates are:
  convert(::Type{T}, ::T) where T at Base.jl:61
Stacktrace:
 [1] _parseany(T::Type, buf::Vector{UInt8}, pos::Int64, len::Int64, opts::Parsers.Options)
   @ CSV C:\Users\gvg\.julia\packages\CSV\jFiCn\src\utils.jl:455
 [2] parsevalue!
   @ C:\Users\gvg\.julia\packages\CSV\jFiCn\src\file.jl:810 [inlined]
 [3] parserow
   @ C:\Users\gvg\.julia\packages\CSV\jFiCn\src\file.jl:620 [inlined]
 [4] parsefilechunk!(ctx::CSV.Context, pos::Int64, len::Int64, rowsguess::Int64, rowoffset::Int64, columns::Vector{CSV.Column}, #unused#::Type{Tuple{}})
   @ CSV C:\Users\gvg\.julia\packages\CSV\jFiCn\src\file.jl:551
 [5] CSV.File(ctx::CSV.Context, chunking::Bool)
   @ CSV C:\Users\gvg\.julia\packages\CSV\jFiCn\src\file.jl:291
 [6] File
   @ C:\Users\gvg\.julia\packages\CSV\jFiCn\src\file.jl:226 [inlined]
 [7] #File#25
   @ C:\Users\gvg\.julia\packages\CSV\jFiCn\src\file.jl:222 [inlined]
 [8] top-level scope
   @ REPL[21]:1
```

**Comment by sairus7** [comment]
The problem is with transition from `InlineStrings@1.2.0` to `InlineStrings@1.2.1`

**Comment by nickrobinson251** [comment]
Sorry about that. Can you try with InlineStrings v1.2.2 ?

**Comment by nickrobinson251** [comment]
If you do `using Pkg; Pkg.update()` hopefully that will now get v1.2.2 and hopefully fix the issue

**Comment by nickrobinson251** [comment]
hmm 🤔 i _should_ have been registered a couple hours ago https://github.com/JuliaRegistries/General/pull/70091

**Comment by nickrobinson251** [comment]
The issue does seem fixed on InlineStrings v1.2.2

If you start Julia with the environment variable `JULIA_PKG_SERVER_REGISTRY_PREFERENCE=eager` then hopefully that'll get you v1.2.2 🤞 
e.g.
```
$> JULIA_PKG_SERVER_REGISTRY_PREFERENCE=eager julia --project -q
(@v1.8) pkg> add InlineStrings@1.2.2
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
    Updating `~/.julia/environments/v1.8/Project.toml`
  [842dd82b] + InlineStrings v1.2.2
    Updating `~/.julia/environments/v1.8/Manifest.toml`
  [842dd82b] + InlineStrings v1.2.2
```

(thanks @giordano for this suggestion via slack)

**Comment by nickrobinson251** [comment]
will close this now, but please comment if you still have any issues - thanks for the report!
